compiler/plugins/target/AMD-AIE/iree-amd-aie/Transforms/KernelDispatch.cpp

```diff
@@ -73,158 +73,71 @@ FailureOr<std::array<uint32_t, 3>> getPackedSize(linalg::LinalgOp linalgOp,
 // the pad-pack approach to tiling a matmul. Also contains the packing sizes for
 // the M, N, and K dimensions. The packing sizes correspond to matmul
 // vector-instruction sizes for vectorizable types.
-class PadPackTiling {
+class ParameterSetting {
  public:
-  SmallVector<int64_t> getPackSize() const { return {mPack, nPack, kPack}; }
+  SmallVector<int64_t> getPackSizeL0() const {
+    return {m0Pack, n0Pack, k0Pack};
+  }
+  SmallVector<int64_t> getPackSizeL1() const {
+    return {m1Pack, n1Pack, k1Pack};
+  }
 
-  static FailureOr<PadPackTiling> create(linalg::LinalgOp linalgOp);
+  static FailureOr<ParameterSetting> create(linalg::LinalgOp linalgOp,
+                                            bool isPackPeel);
 
   uint32_t getM0() const { return M0; }
   uint32_t getN0() const { return N0; }
   uint32_t getK0() const { return K0; }
   uint32_t getM1() const { return M1; }
   uint32_t getN1() const { return N1; }
   uint32_t getK1() const { return K1; }
-  uint32_t getMPack() const { return mPack; }
-  uint32_t getNPack() const { return nPack; }
-  uint32_t getKPack() const { return kPack; }
+  uint32_t getM0Pack() const { return m0Pack; }
+  uint32_t getN0Pack() const { return n0Pack; }
+  uint32_t getK0Pack() const { return k0Pack; }
+  uint32_t getM1Pack() const { return m1Pack; }
+  uint32_t getN1Pack() const { return n1Pack; }
+  uint32_t getK1Pack() const { return k1Pack; }
 
  private:
-  PadPackTiling(uint32_t M0, uint32_t N0, uint32_t K0, uint32_t M1, uint32_t N1,
-                uint32_t K1, uint32_t mPack, uint32_t nPack, uint32_t kPack)
+  ParameterSetting(uint32_t M0, uint32_t N0, uint32_t K0, uint32_t M1,
+                   uint32_t N1, uint32_t K1, uint32_t m0Pack, uint32_t n0Pack,
+                   uint32_t k0Pack, uint32_t m1Pack, uint32_t n1Pack,
+                   uint32_t k1Pack)
       : M0(M0),
         N0(N0),
         K0(K0),
         M1(M1),
         N1(N1),
         K1(K1),
-        mPack(mPack),
-        nPack(nPack),
-        kPack(kPack) {
-    // Tiling at level 0 (shared memory) should be no smaller that tiling at
-    // level 1 (AIE core memory).
-    assert(M0 >= M1 && N0 >= N1);
-
-    // Tiling at level 1 (AIE core memory) should be no smaller than the packing
-    // size (vector instruction size).
-    assert(M1 >= mPack && N1 >= nPack);
-  }
+        m0Pack(m0Pack),
+        n0Pack(n0Pack),
+        k0Pack(k0Pack),
+        m1Pack(m1Pack),
+        n1Pack(n1Pack),
+        k1Pack(k1Pack) {}
 
   uint32_t M0;
   uint32_t N0;
   uint32_t K0;
   uint32_t M1;
   uint32_t N1;
   uint32_t K1;
-  uint32_t mPack;
-  uint32_t nPack;
-  uint32_t kPack;
+  uint32_t m0Pack;
+  uint32_t n0Pack;
+  uint32_t k0Pack;
+  uint32_t m1Pack;
+  uint32_t n1Pack;
+  uint32_t k1Pack;
 };
 
-FailureOr<PadPackTiling> PadPackTiling::create(linalg::LinalgOp linalgOp) {
+FailureOr<ParameterSetting> ParameterSetting::create(linalg::LinalgOp linalgOp,
```

Comment:
Another design (maybe neater, maybe not...) would be to have 2 create methods

createForPadPack(...)
createForPackPeel(...)